src/main/java/io/kuzzle/sdk/security/KuzzleSecurity.java

```diff
@@ -0,0 +1,1044 @@
+   */
+  public void getRole(@NonNull final String id, KuzzleOptions options, @NonNull final KuzzleResponseListener<KuzzleRole> listener) {
+    JSONObject data;
+
+    if (id == null) {
+      throw new IllegalArgumentException("KuzzleSecurity.getRole: a role ID is required");
+    }
+
+    if (listener == null) {
+      throw new IllegalArgumentException("KuzzleSecurity.getRole: a listener is required");
+    }
+
+    try {
+      data = new JSONObject().put("_id", id);
+      this.kuzzle.query(buildQueryArgs("getRole"), data, options, new OnQueryDoneListener() {
+        @Override
+        public void onSuccess(JSONObject response) {
+          try {
+            JSONObject result = response.getJSONObject("result");
+            listener.onSuccess(new KuzzleRole(KuzzleSecurity.this.kuzzle, result.getString("_id"), result.getJSONObject("_source")));
+          }
+          catch (JSONException e) {
+            throw new RuntimeException(e);
+          }
+        }
+
+        @Override
+        public void onError(JSONObject error) {
+          listener.onError(error);
+        }
+      });
+    } catch (JSONException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Retrieves a single Role using its unique Role ID
+   *
+   * @param id - unique role ID
+   * @param listener - response callback
+   */
+  public void getRole(@NonNull final String id, @NonNull final KuzzleResponseListener<KuzzleRole> listener) {
+    getRole(id, null, listener);
+  }
+
+
+  /**
+   * Executes a search on roles using a set of filters
+   *
+   *  * /!\ There is a small delay between role creation and their existence in our persistent search layer,
+   * usually a couple of seconds.
+   * That means that a role that was just been created won’t be returned by this function.
+   *
+   * @param filters - search filters (see ElasticSearch filters)
+   * @param options - Optional query arguments
+   * @param listener - Callback listener
+   * @throws JSONException
+   */
+  public void searchRoles(@NonNull final JSONObject filters, final KuzzleOptions options, @NonNull final KuzzleResponseListener<KuzzleSecurityDocumentList> listener) throws JSONException {
+    if (filters == null) {
+      throw new IllegalArgumentException("KuzzleSecurity.searchRoles: filters cannot be null");
+    }
+
+    if (listener == null) {
+      throw new IllegalArgumentException("KuzzleSecurity.searchRoles: a callback listener is required");
+    }
+
+    JSONObject data = new JSONObject().put("body", filters);
+
+    this.kuzzle.query(buildQueryArgs("searchRoles"), data, options, new OnQueryDoneListener() {
+      @Override
+      public void onSuccess(JSONObject response) {
+        try {
+          JSONObject result = response.getJSONObject("result");
+          JSONArray documents = result.getJSONArray("hits");
+          int documentsLength = documents.length();
+          LinkedList<AbstractKuzzleSecurityDocument> roles = new LinkedList<>();
```

Comment:
To be honest, I took the 1st implementation of the `List` interface I could find :-)
`ArrayList` seems a bit better because `get` is constant-time. I'll change this ASAP, since it has no real impact on the code.